src/mango/src/mango_execution_stats.erl

```diff
@@ -37,12 +37,23 @@
 -define(SHARD_STATS_KEY, mango_shard_execution_stats).
 
 to_json(Stats) ->
+    to_json(Stats, true).
+
+to_json(Stats, IncludeDbName) ->
+    Base =
```

Comment:
Or you could just simply say:

```erlang
Base = [{dbname, Stats#execution_stats.dbname} || IncludeDbName]
```

Reply:
I'm not quite following your suggestion, we need to include the tuple `{dbname, Stats#execution_stats.dbname}` in the list conditionally. I think perhaps you're suggesting to move the construction of the the base into the arity one function head, like:

```
to_json(Stats) ->
    to_json(Stats, [{dbname, Stats#execution_stats.dbname}]).

to_json(Stats, Base) ->
  {[
    ...
    | Base
  ]}.
```

But that would change the caller here https://github.com/apache/couchdb/pull/4990/files/69f80aa3a270030601ebd111bd3a1dc62a5df8fa#diff-478704bf59b4beb680fc69863dfe63a544cfea5d6dca7507d6ce080c84b6cb35R119 to have to supply the base list into `to_json`, eg `to_json(Stats, [])` instead of `to_json(Stats, false)`, and I don't think having to supply the base list instead of a boolean is a cleaner approach.

Is this what you were suggesting or something else? I agree that it's a bit clunky for sure.

Reply:
There is no need to change the function's interface.  What I suggested is a semantical equivalent of your original definition.  This alternative syntax is based on how [list comprehensions in Erlang](https://www.erlang.org/doc/programming_examples/list_comprehensions) work.  The Boolean value on the right-hand side of the `||` symbol acts as a predicate and the value on the left-hand side is added to the resulting list if that is `true` otherwise omitted.

For example, consider the following in the Erlang shell:

```
> [1 || true].
[1]
> [1 || false].
[]
```

List comprehensions do not require the use of variables, hence this formulation becomes a valid although unusual instance.

Reply:
Ahhhhh I see what you mean, I haven't seen that structure before. Interesting, but seems a bit esoteric. Perhaps if we make that a common pattern in the codebase.

Reply:
That said, it would look nice in this case as well.  Of course, this is just a suggestion, I am adding this example for consideration only where `Base` is removed completely:

```erlang
   {[
        {total_keys_examined, Stats#execution_stats.totalKeysExamined},
        {total_docs_examined, Stats#execution_stats.totalDocsExamined},
        {total_quorum_docs_examined, Stats#execution_stats.totalQuorumDocsExamined},
        {results_returned, Stats#execution_stats.resultsReturned},
        {execution_time_ms, Stats#execution_stats.executionTimeMs}
        | [{dbname, Stats#execution_stats.dbname} || IncludeDbName]
    ]}.
```